### PR TITLE
Playground fixes and improvements

### PR DESCRIPTION
**Pull request by shmish111**
Since we are moving fast, this PR gives a lot of small changes, it's best to read the commit messages for an idea of what is in this PR.

### CODE REVIEW

File: plutus-playground/plutus-playground-client/src/StaticData.purs [hunk 1]
```diff
@@ -2,162 +2,19 @@ module StaticData
...
+import Data.Map (Map)
+import Data.Map as Map
+import Data.Tuple.Nested ((/\))
+import Playground.Usecases (vesting, game, crowdfunding, messages)
+
+type Label = String
+type Contents = String
+
+editorContents :: Map Label Contents
+editorContents =
```
Comment (j-mueller) [inline plutus-playground/plutus-playground-client/src/StaticData.purs]: 👏

File: plutus-playground/plutus-playground-server/app/PSGenerator.hs [hunk 2]
```diff
@@ -176,3 +212,9 @@ generate outputDir = do
         myBridgeProxy
         (Proxy @API.API)
     writePSTypes outputDir (buildBridge myBridge) myTypes
+    writeUsecases outputDir
+
+k :: IO ()
+k =
+    generate
+        "/Users/kris/Work/Clients/IOHK/plutus/plutus-playground/plutus-playground-client/src"
```
Comment (j-mueller) [inline plutus-playground/plutus-playground-server/app/PSGenerator.hs]: That looks like it will only work on one machine, and also what is `k` ? `kris`?

Comment (shmish111) [reply plutus-playground/plutus-playground-server/app/PSGenerator.hs]: deleted

**Comment by j-mueller** [comment]
Fine, but some of the commits only fix things that were introduced by earlier commits in the same PR. Example: 181e284 and 82765ed. Ideally the build would succeed at each commit.

File: plutus-playground/plutus-playground-server/plutus-playground-server.cabal [hunk 3]
```diff
@@ -26,6 +26,7 @@ library
   exposed-modules:
       Playground.Server
       Playground.Interpreter
+      Playground.Usecases
```
Comment (j-mueller) [inline plutus-playground/plutus-playground-server/plutus-playground-server.cabal]: You should probably add the sources for those under `extra-source-files` (that might be causing the build to fail as well)

File: plutus-playground/plutus-playground-server/src/Playground/Usecases.hs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Playground.Usecases where
+
+import           Data.ByteString (ByteString)
+import           Data.FileEmbed  (embedFile)
+
+vesting :: ByteString
+vesting = $(embedFile "./usecases/Vesting.hs")
```
Comment (michaelpj) [inline plutus-playground/plutus-playground-server/src/Playground/Usecases.hs]: This is great, but why are we taking copies of these thing? I know Jann is planning to write simplified versions at some point, but until then shouldn't we use the ones from `plutus-use-cases`?

**Comment by michaelpj** [comment]
> Ideally the build would succeed at each commit.

That's the aspiration, but it's hard to live up to...

File: plutus-playground/plutus-playground-server/src/Playground/Usecases.hs (hunk 4 above)
Comment (michaelpj) [reply plutus-playground/plutus-playground-server/src/Playground/Usecases.hs]: Oh I see, they're quite different due to the pragmas now.

File: plutus-playground/plutus-playground-server/plutus-playground-server.cabal (hunk 3 above)
Comment (shmish111) [reply plutus-playground/plutus-playground-server/plutus-playground-server.cabal]: this didn't work but there is http://hackage.haskell.org/package/file-embed-0.0.11/docs/Data-FileEmbed.html#v:makeRelativeToProject which seems to work

**Comment by shmish111** [comment]
@j-mueller we can squash if you like, the thing is that the different commits have different subjects so a squash will lose a lot of context

**Comment by j-mueller** [comment]
> we can squash if you like, the thing is that the different commits have different subjects so a squash will lose a lot of context

Nah I think in this case it's better to keep them as they are. The compilation error is probably more important.

File: plutus-playground/plutus-playground-server/plutus-playground-server.cabal (hunk 3 above)
Comment (j-mueller) [reply plutus-playground/plutus-playground-server/plutus-playground-server.cabal]: nice

**Comment by shmish111** [comment]
Unfortunately I can't get tests that use the interpreter working with Nix at the moment and it's taking too long to investigate so for now I've commented them out. They work with stack/cabal so you can manually run them by uncommenting them

**Comment by michaelpj** [comment]
I bet I know what it is, give me a sec.

File: plutus-playground/plutus-playground-server/src/Playground/Usecases.hs (hunk 4 above)
Comment (j-mueller) [reply plutus-playground/plutus-playground-server/src/Playground/Usecases.hs]: they also use `mkFunction` to generate the schema, and that doesn't exist in `plutus-use-cases`

**Comment by michaelpj** [comment]
And it's green!

**Pull request merged**